examples/dump_torrent.cpp

```diff
@@ -58,40 +58,37 @@ std::vector<char> load_file(std::string const& filename)
 
 int main(int argc, char* argv[]) try
 {
-	if (argc < 2 || argc > 4) {
-		std::cerr << "usage: dump_torrent torrent-file [total-items-limit] [recursion-limit]\n";
+	if (argc < 2 || argc > 5) {
+		std::cerr << "usage: dump_torrent torrent-file [total-items-limit] [recursion-limit] [piece-count-limit]\n";
 		return 1;
 	}
 
-	int item_limit = 1000000;
-	int depth_limit = 1000;
+	lt::load_torrent_limits cfg;
 
-	if (argc > 2) item_limit = atoi(argv[2]);
-	if (argc > 3) depth_limit = atoi(argv[3]);
+	if (argc > 2) cfg.max_decode_tokens = atoi(argv[2]);
+	if (argc > 3) cfg.max_decode_depth = atoi(argv[3]);
+	if (argc > 4) cfg.max_pieces = atoi(argv[4]);
 
 	std::vector<char> buf = load_file(argv[1]);
-	lt::bdecode_node e;
 	int pos = -1;
 	lt::error_code ec;
-	std::cout << "decoding. recursion limit: " << depth_limit
-		<< " total item count limit: " << item_limit << "\n";
-	int const ret = lt::bdecode(&buf[0], &buf[0] + buf.size(), e, ec, &pos
-		, depth_limit, item_limit);
+	std::cout << "decoding. recursion limit: " << cfg.max_decode_depth
+		<< " total item count limit: " << cfg.max_decode_tokens << "\n";
+	lt::bdecode_node const e = lt::bdecode(buf, ec, &pos, cfg.max_decode_depth
```

Comment:
Can you please explain the reason why `cfg.max_pieces` is also not passed here, like the others? Why is it not a field in `bdecode_node`?

For example, in `torrent_info:1376`, it is also passed separately:
```
bool torrent_info::parse_torrent_file(bdecode_node const& torrent_file
		, error_code& ec, int const piece_limit)
```

Reply:
this is a call to a different function, `bdecode()`. All this function does is to parse the bdecoded structure. It doesn't do anything related to the number of pieces, just the recursion depth and number of tokens to parse.

The reason this example, `dump_torrent`, calls `bdecode()` itself, instead of just relying on the higher level `torrent_info` constructor, is that it also prints the raw bdecoded file first. In a way this is unnecessary, but it's really useful for debugging stuff. I should probably separate that out to a separate program though.

Reply:
Ok, thanks for the explanation